<!-- Launch file for the rrt-detector and the assigner -->


<launch>
  <arg name="eta" value="1.0"/>
  <arg name="Geta" value="1.0"/>
  <param name="namespace_init_count" value="1"/>


  <node pkg="rrt_exploration" type="local_rrt_detector" name="robot1_rrt_detector" output="screen">
    <param name="eta" value="$(arg eta)"/>
    <param name="map_topic" value="map_merge/map"/>
    <param name="robot_frame" value="robot1/base_link"/>
    <param name="rate" value="100"/>
  </node>

  <!-- <node pkg="rrt_exploration" type="robot_assigner" name="$(arg namespace)_assigner" output="screen">
    <param name="map_topic" value="$(arg namespace)/map"/>
    <param name="costmap_topic" value="$(arg namespace)/move_base_node/global_costmap/costmap"/>
    <param name="robot_frame" value="$(arg namespace)/map"/>
    <param name="info_radius" value="1"/> 
    <param name="costmap_pixel_threshold" value="0"/> 
    <param name="frontier_topic" value="$(arg namespace)/detected_points"/>
    <param name="namespace" value="$(arg namespace)"/> 
    <param name="n_robots" value="1"/>
    <param name="rate" value="5"/>
  </node> -->
 
 <node pkg="rrt_exploration" type="filter_jianming.py" name="filter" output="screen">
    <param name="map_topic" value="map_merge/map"/>
    <param name="costmap_topic" value="map_merge/costmap"/>
    <param name="frontier_topic" value="detected_points"/>
    <param name="n_robots" value="1"/>
    <param name="rateHZ" value="4"/>
    <param name="namespace" value="robot"/>
    <param name="robot_frame" value="base_link"/>
    <param name="info_radius" value="1.0"/> 
    <param name="costmap_pixel_threshold" value="20"/> 
  </node>


  <node pkg="rrt_exploration" type="assigner_jianming_nocallfunction.py" name="assigner" output="screen">
    <param name="global_frame" value="robot1/map"/>
    <param name="robot_frame_suffix" value="/map"/>
    <param name="map_topic" value="map_merge/map"/>
    <param name="frontiers_topic" value="filtered_points"/> 
    <param name="n_robots" value="1"/>
    <param name="namespace" value="robot"/>
    <param name="namespace_init_count" value="1"/>
    <param name="rate" value="4"/>
    <param name="mini_step" value="0.5"/>
    <param name="info_radius" value="1.0"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rrt_exploration_tutorials)/launch/includes/rviz_config/single_python_multi_nodes.rviz">
    <remap from="move_base_simple/goal" to="robot1/move_base_simple/goal"/>
  </node>

  <node pkg="mapmerge" type="mapmerge_node" name="map_merge" output="screen" >
    <param name="robot1_x" value="0.0" />
    <param name="robot1_y" value="0.0" />
    <param name="resolution" value="0.1" />
    <param name="map_width" value="500" />
    <param name="map_height" value="500" />
  </node>

  <node pkg="mapmerge" type="mapmerge_costmap_node" name="map_merge_costmap" output="screen" >
	<param name="robot1_x" value="0.0" />
	<param name="robot1_y" value="0.0" />
	<param name="resolution" value="0.05" />
	<param name="map_width" value="3000" />
	<param name="map_height" value="3000" />
  <param name="robot1costmap" value="/robot1/move_base_node/global_costmap/costmap" />
</node>

</launch>



